### PR TITLE
Don't use Apache init scripts

### LINKED ISSUE #1638: letsencrypt stops apache, then fails
Relevant log except:

```
2015-11-30 21:18:48,589:INFO:letsencrypt.reporter:Reporting to user: The following 'urn:acme:error:connection' errors were reported by the server
:

Domains: roundcube.singpolyma.net
Error: The server could not connect to the client for DV
2015-11-30 21:18:48,614:INFO:letsencrypt.auth_handler:Cleaning up challenges
2015-11-30 21:18:48,782:ERROR:letsencrypt_apache.configurator:Apache Reload Failed!
Reloading web server: apache2 failed!
Apache2 is not running ... (warning).
```

After which apache is not running.  It was running before I ran letsencrypt (running from master, latest) and starting it again works fine.  It seems that letsencrypt is stopping it and then complaining that it is stopped...

### PR DESCRIPTION
**Pull request by bmw**
Partially resolves #1638. PR removes reliance on Debian specific Apache init scripts, restores the ability to run `letsencrypt --apache` if Apache isn't currently running, and adds utility function for registering deprecated command line flags.

### CODE REVIEW

**Comment by bmw** [comment]
Test failures should be fixed by letsencrypt/boulder#1214.

File: letsencrypt/le_util.py [hunk 1]
```diff
@@ -255,3 +257,26 @@ def safe_email(email):
     else:
         logger.warn("Invalid email address: %s.", email)
         return False
+
+
+def add_deprecated_argument(add_argument, argument_name, nargs):
```
Comment (pde) [inline letsencrypt/le_util.py]: It feels like this should be part of HelpfulArgumentParser or at least cli.py, rather than le_util.

Comment (pde) [inline letsencrypt/le_util.py]: (though I'm fine throwing that task in the "refactor" pile for later)

**Comment by pde** [comment]
While testing this on Ubuntu 14.04:

```
Traceback (most recent call last):
  File "venv/bin/letsencrypt", line 9, in <module>
    load_entry_point('letsencrypt==0.1.0.dev0', 'console_scripts', 'letsencrypt')()
  File "/home/ubuntu/letsencrypt/letsencrypt/cli.py", line 1225, in main
    return args.func(args, config, plugins)
  File "/home/ubuntu/letsencrypt/letsencrypt/cli.py", line 451, in run
    installer, authenticator = choose_configurator_plugins(args, config, plugins, "run")
  File "/home/ubuntu/letsencrypt/letsencrypt/cli.py", line 421, in choose_configurator_plugins
    authenticator = installer = display_ops.pick_configurator(config, req_inst, plugins)
  File "/home/ubuntu/letsencrypt/letsencrypt/display/ops.py", line 123, in pick_configurator
    (interfaces.IAuthenticator, interfaces.IInstaller))
  File "/home/ubuntu/letsencrypt/letsencrypt/display/ops.py", line 80, in pick_plugin
    verified.prepare()
  File "/home/ubuntu/letsencrypt/letsencrypt/plugins/disco.py", line 213, in prepare
    return [plugin_ep.prepare() for plugin_ep in self._plugins.itervalues()]
  File "/home/ubuntu/letsencrypt/letsencrypt/plugins/disco.py", line 103, in prepare
    self._initialized.prepare()
  File "/home/ubuntu/letsencrypt/letsencrypt-apache/letsencrypt_apache/configurator.py", line 142, in prepare
    if not le_util.exe_exists(exe):
  File "/home/ubuntu/letsencrypt/letsencrypt/le_util.py", line 74, in exe_exists
    path, _ = os.path.split(exe)
  File "/home/ubuntu/letsencrypt/venv/lib/python2.7/posixpath.py", line 92, in split
    i = p.rfind('/') + 1
AttributeError: 'NoneType' object has no attribute 'rfind'
```

File: letsencrypt/le_util.py (hunk 1 above)
Comment (bmw) [inline letsencrypt/le_util.py]: I'd argue that it should not go in `HelpfulArgumentParser` as then you'd need access to the parser object to be able to call the function. This approach gives 3rd party plugins a function they can use to deprecate their own arguments.

If we split `cli.py` into `main.py` and `cli.py` as you have suggested, I think putting it in `cli.py` makes sense. I personally like it best in `le_util.py` based on the current structure though.

**Comment by bmw** [comment]
Oh man. Thanks for catching that. My last commit before submitting this PR was supposed to only modify the unit tests but one of the changes I made broke everything.

Problem has been fixed. Ready for re-review.

**Comment by pde** [comment]
Now LGTM.

**Pull request merged**